### ISSUE TITLE
Incorrect urls returned for category pages on multi-store instance

### ISSUE THREAD

**Issue by gjportegies** [elgentos/magento2-alternate-urls#5]
I was testing this module on one of our shops, which is a multi-store instance, and found that on category pages the same Url is returned for all languages/stores.
The shop has the following languages configured: NL, DE, EN and FR.
The configuration setting 'Add Store Code to Urls' is set to 'Yes'.

<img width="543" alt="Screen Shot 2022-08-16 at 10 24 14" src="https://user-images.githubusercontent.com/6707410/184834325-36e5a92d-5974-4030-b7fa-c31c4ea29a6a.png">

I made some quick changes in `Type\Category.php` to get it working. It needs some rethinking on naming etc., so I've not created a PR yet. Changed are based on this Stackoverflow answer https://magento.stackexchange.com/a/283342/45764

```diff
<?php

declare(strict_types=1);

namespace Elgentos\AlternateUrls\Type;

use Elgentos\AlternateUrls\Model\AlternateUrl;
use Magento\Catalog\Api\CategoryRepositoryInterface;
use Magento\Catalog\Api\Data\CategoryInterface;
use Magento\Catalog\Model\Category as CategoryModel;
use Magento\Framework\App\Config\ScopeConfigInterface;
use Magento\Framework\App\RequestInterface;
use Magento\Framework\Escaper;
use Magento\Framework\Exception\NoSuchEntityException;
use Magento\Framework\Registry;
use Magento\Framework\Serialize\Serializer\Json;
use Magento\Framework\View\Element\Block\ArgumentInterface;
use Magento\Store\Model\StoreManagerInterface;
+ use Magento\CatalogUrlRewrite\Model\CategoryUrlPathGenerator;

class Category extends AbstractType implements TypeInterface, ArgumentInterface
{
    private ?CategoryInterface $currentCategory = null;

    private Registry $registry;

    private CategoryRepositoryInterface $categoryRepository;

+   private CategoryUrlPathGenerator $categoryUrlPathGenerator;

    public function __construct(
        Json $serializer,
        ScopeConfigInterface $scopeConfig,
        StoreManagerInterface $storeManager,
        RequestInterface $request,
        Registry $registry,
        CategoryRepositoryInterface $categoryRepository,
        CategoryUrlPathGenerator $categoryUrlPathGenerator
    ) {
        parent::__construct(
            $serializer,
            $scopeConfig,
            $storeManager,
            $request
        );

        $this->registry                 = $registry;
        $this->categoryRepository       = $categoryRepository;
+       $this->categoryUrlPathGenerator = $categoryUrlPathGenerator;
    }

    public function getAlternateUrls(): array
    {
        $currentCategory = $this->getCurrentCategory();

        if (!$currentCategory || !$currentCategory->getId()) {
            return [];
        }

        $result = [];

        foreach ($this->getMapping() as $item) {
            try {
                $result[] = new AlternateUrl(
                    $item['hreflang'],
-                    $this->getStoreCategory((int) $item['store_id'])->getUrl()
+                    $this->getUrl((int) $item['store_id'])
                );
            } catch (NoSuchEntityException $e) {
                continue;
            }
        }

        return $result;
    }

    public function getCurrentCategory(): ?CategoryInterface
    {
        return $this->currentCategory ??= $this->registry->registry('current_category');
    }

    /**
     * @throws NoSuchEntityException
     */
    private function getStoreCategory(
        int $storeId
    ): CategoryModel {
        $currentStore    = $this->storeManager->getStore();
        $currentCategory = $this->getCurrentCategory();

        /** @var CategoryModel $category */
        $category = $currentStore->getId() === $storeId
            ? $currentCategory
            : $this->categoryRepository->get(
                $currentCategory->getId(),
                $storeId
            );

        return $category;
    }

+    public function getUrl($storeId): string
+    {
+        $store = $this->storeManager->getStore($storeId);
+        $category = $this->getStoreCategory($storeId);
+
+        $path = $this->categoryUrlPathGenerator->getUrlPathWithSuffix($category);
+        return $store->getBaseUrl() . $path;
+    }
}

```

The result after the changes:

<img width="460" alt="Screen Shot 2022-08-16 at 10 50 44" src="https://user-images.githubusercontent.com/6707410/184838832-97483ce9-8365-4ddc-b1a2-cc9dd8096778.png">

**Comment by ArjenMiedema** [comment]
@gjportegies thanks for sharing this. I'll have a look at it and will create a fix.

**Comment by ArjenMiedema** [comment]
This has been fixed in release 1.1.1. Please upgrade to that version for the fixes to be available.

Thanks for sharing the issue and the fix.
